src/main/resources/io/viash/platforms/nextflow/WorkflowHelper.nf

```diff
@@ -603,10 +606,467 @@ def paramsToList(params, config) {
+  def paramListFile = paramListOut[0]
+  def paramSets = paramListOut[1] // these are the actual parameters from reading the blob/file
+
+  // data checks
+  assert paramSets instanceof List: "--param_list should contain a list of maps"
+  for (value in paramSets) {
+    assert value instanceof Map: "--param_list should contain a list of maps"
+  }
+
+  // Reformat from List<Map> to List<Tuple2<String, Map>> by adding the ID as first element of a Tuple2
+  paramSets = paramSets.collect({ paramValues ->
+    [paramValues.get("id", null), paramValues.findAll{it.key != 'id'}]
+  })
+  // Split parameters with 'multiple: true'
+  paramSets = paramSets.collect({ paramSet ->
+    def id = paramSet[0]
+    def paramValues = paramSet[1]
+    def splitParamValues = _splitParams(paramValues, multiArguments)
+    [id, splitParamValues]
+  })
+  
+  // The paths of input files inside a param_list file may have been specified relatively to the
+  // location of the param_list file. These paths must be made absolute.
+  if (paramListFile){
+    paramSets = paramSets.collect({ paramSet ->
+      def id = paramSet[0]
+      def paramValues = paramSet[1]
+      def relativeParamValues = _resolvePathsRelativeTo(paramValues, inputfileArguments, paramListFile)
+      [id, relativeParamValues]
+    })
+  }
+
+  return paramSets
+}
+
+/**
+ * Cast parameters to the correct type as defined in the Viash config.
+ */
+
+private Map<String, Object> _castParamTypes(Map<String, Object> parValues, Map config) {
+  def configArguments = config.functionality.allArguments
+    // Cast the input to the correct type according to viash config
+  def castParValues = parValues.collectEntries({ parName, parValue ->
+    paramSettings = configArguments.find({it.plainName == parName})
+    // dont parse parameters like publish_dir ( in which case paramSettings = null)
+    parType = paramSettings ? paramSettings.get("type", null) : null
+    if (! (parValue instanceof Collection)) {
+      parValue = [parValue]
+    }
+    if (parType == "file" && ((paramSettings.direction ?: "input") == "input")) {
+      parValue = parValue.collect{ path ->
+        if (path !instanceof String) {
+          path
+        } else {
+          file(path)
+        }
+      }
+    } else if (parType == "integer") {
+      parValue = parValue.collect{it as Integer}
+    } else if (parType == "double") {
+      parValue = parValue.collect{it as Double}
+    } else if (parType == "boolean" || 
+                parType == "boolean_true" || 
+                parType == "boolean_false") {
+      parValue = parValue.collect{it as Boolean}
+    }
+
+    // simplify list to value if need be
+    if (paramSettings && !paramSettings.multiple) {
+      assert parValue.size() == 1 : 
+        "Error: argument ${parName} has too many values.\n" +
+        "  Expected amount: 1. Found: ${parValue.size()}"
+      parValue = parValue[0]
+    }
```

Comment:
This transformation is happening twice; here and in `_splitParams`. Is this intentional?

Reply:
I was needed before we applied `applyConfig` to everything, but not anymore.